### PR TITLE
pdns: rename from powerdns

### PR DESCRIPTION
**Pull request by mweinelt**
The proper name for this package should be taken from their package
repository¹.

<s>This will likely reenable updates by @r-ryantm, because repology² also
uses that name.</s>

[1] https://github.com/PowerDNS/pdns
[2] https://repology.org/project/pdns/versions

###### Description of changes

I was wondering why our package was so severly outdated, when I noticed that r-ryantm stopped updating it a while ago. After talking to ryantm on matrix¹ the conclusion was that nixpkgs-update does not handle derivations with multiple hashes, so removing the patch in the 4.6.1 update already fixed the problem.

[1] https://matrix.to/#/!kjdutkOsheZdjqYmqp:nixos.org/$bnJRaScKapuBqiOscouBOp6uuScUt_bSJVjgsfbM28A?via=nixos.org&via=matrix.org&via=nixos.dev

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by NickCao — APPROVED** [review]
This also make the package name in line with the binary names, a great improvement.

**Comment by NickCao** [comment]
Not expecting r-ryantm to bail out on that simple case, but anyway.

> @NickCao Updated the OP. The reason r-ryantm didn't update pdns was that it had two hashes (src + patch). Updating to 4.6.1 and dropping the patch should have fixed that.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -23226,7 +23226,7 @@ with pkgs;
 
   semodule-utils = callPackage ../os-specific/linux/semodule-utils { };
 
-  powerdns = callPackage ../servers/dns/powerdns {
+  pdns = callPackage ../servers/dns/pdns {
     boost = boost172;
   };
```
[the remark this answers is not in the document]
Comment (NickCao) [reply pkgs/top-level/all-packages.nix:23231]: A remnant from the old versions I guess?

[the remark this answers is not in the document]
Comment (mweinelt) [reply pkgs/top-level/all-packages.nix:23231]: Probably, I'll add a commit in a bit.

**Pull request merged**